Sources/XCMetricsUtils/Shell.swift

```diff
@@ -74,9 +74,34 @@ private func shellInternal(_ cmd: String, args: [String] = [], stdout: PipeLike?
     task.launch()
     task.waitUntilExit()
     if task.terminationStatus != 0 {
+        if stderr != nil {
+            // Error stream was captured so cannot inspect its content
+            throw ShellError.statusError("Failed command", task.terminationStatus)
+        }
         let errorData = errorHandle.fileHandleForReading.readDataToEndOfFile()
-        let errorString = String(data: errorData, encoding: .utf8)?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "No error returned from the process."
+        let errorString = String(data: errorData, encoding: .utf8)?.trim() ?? "No error returned from the process."
         throw ShellError.statusError(
-            "status \(task.terminationStatus): \(errorString)", task.terminationStatus)
+            "status \(task.terminationStatus): \(errorString)", task.terminationStatus
+        )
+    }
+}
+
+extension String {
+    public func trim() -> String {
+        func trim(_ separator: String) -> String {
+            var E = endIndex
+            while String(self[startIndex..<E]).hasSuffix(separator) && E > startIndex {
+                E = index(before: E)
+            }
+            return String(self[startIndex..<E])
+        }
+
+        if hasSuffix("\r\n") {
```

Comment:
Interesting, so the output from `pmset` returned `\r` and `\n\`?

Reply:
This part I just copied from an internal project we have, in order to have `Shell.swift` ideally the same everywhere.